lib/setupHooks/installFromCargoBuildLogHook.sh

```diff
@@ -13,12 +13,17 @@ function installFromCargoBuildLog() (
   logs=$(jq -R 'fromjson?' <"${log}")
 
   local select_non_test='select(.reason == "compiler-artifact" and .profile.test == false)'
-  local select_bins="${select_non_test} | .executable | select(.!= null)"
+  local select_non_dep='select(contains("/deps/")| not)'
```

Comment:
Played around with the bindeps feature myself and I found out that (fortunately) cargo will emit different log entries if a workspace target happens to _also_ be used as a bindep for another member. This means that we can always unconditionally ignore anything in `.../deps/artifact/...`, but we'll still install the actual targets if they happened to be part of the `cargo build` invocation, so modulo some comments here I'm happy to merge this in!